### ISSUE TITLE
Full fix for 500 page when sorting after filtering by tags

### ISSUE THREAD

**Issue by trichoplax** [codidact/qpixel#1053]
**Describe the bug**
[If there are filters that include tags or exclude tags, pressing a sort button gives a 500 error page](https://meta.codidact.com/posts/288150)

**To Reproduce**
1. Go to the posts page for any community, any category
2. Filter to include or exclude 1 or more tags
3. Press a sort button ("Age", "Score", or "Random")
4. See a 500 error page

**Expected behavior**
The filtered posts list should sort into the requested order, with no 500 error.

**Additional context**
I've added [my impressions of the root cause](https://meta.codidact.com/posts/288150/288151#answer-288151) in an answer to the Meta post, and suggested 2 potential fixes. This issue is for implementing the second suggested fix, which may take longer but will use the URL parameters as originally intended. I've raised #1052 for a quick temporary fix that will continue to use the unintended duplicated URL parameters but remove the 500 error for the user.

The full fix is:

When a sort button is clicked, make the URL end with:

```text
&exclude_tags=["tag_id1", "tag_id2"]&sort=score
```

rather than:

```text
&exclude_tags[]=tag_id1&exclude_tags[]=tag_id2&sort=score
```

Similarly for `include_tags`.

This is the reversal of the temporary fix in #1052.

To avoid reintroducing the 500 error, this will require also changing the URL used when there is no sort applied, to end with:

```text
&exclude_tags=["tag_id1", "tag_id2"]
```

rather than:

```text
&exclude_tags[]=tag_id1&exclude_tags[]=tag_id2
```

This will also currently give a 500 error, so the code that displays the posts page will need to be fixed to correctly interpret URL parameters that use comma separated lists of tag ids. This will then remove the problem for the posts page both before and after sorting.

**Comment by trichoplax** [comment]
Thanks for the education. Does this mean you would consider #1052 to be the correct full fix, rather than the quick temporary fix I initially conceived it as?

I couldn't find my way around the HTML spec well enough to determine which of the following is preferable:

```text
&exclude_tags=tag_id1&exclude_tags=tag_id2
```

```text
&exclude_tags[]=tag_id1&exclude_tags[]=tag_id2
```

That is, whether we should have empty square brackets after the array name. If we should, then it seems #1052 may be all we need. If instead they should be omitted, then we may need to change more places.

**Comment by trichoplax** [comment]
No need for apologies. Thank you both for helping me learn.

Is there any further decision needed on this issue or can it be left to be closed by #1146?
